ostd/src/sync/rwlock.rs

```diff
@@ -505,8 +522,11 @@ impl<T: ?Sized + fmt::Debug, R: Deref<Target = RwLock<T, G>> + Clone, G: Guardia
 
 /// A guard that provides immutable data access but can be atomically
 /// upgraded to `RwLockWriteGuard`.
-pub struct RwLockUpgradeableGuard_<T: ?Sized, R: Deref<Target = RwLock<T, G>> + Clone, G: Guardian>
-{
+pub struct RwLockUpgradeableGuard_<
+    T: ?Sized,
+    R: Deref<Target = RwLock<T, G>> + Clone,
+    G: SpinGuardian,
+> {
```

Comment:
Looks like you don't implement `AsAtomicModeGuard` for the upgradeable guard. Any reason? We should implement it for consistency.